packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+  bool _atLineStart = true;
+  bool _hasMultipleLines = false;
+
+  /// Whether the string being built already has more than 1 line.
+  bool get hasMultipleLines => _hasMultipleLines;
+
+  /// Write text ensuring the specified prefixes for the first and subsequent
+  /// lines.
+  void write(String s) {
+    if (s.isEmpty)
+      return;
+
+    if (s == '\n') {
+      // Edge case to avoid adding trailing whitespace when the caller did
+      // not explicitly add trailing trailing whitespace.
+      if (_buffer.isEmpty) {
+        _buffer.write(prefixLineOne.trimRight());
+      } else if (_atLineStart) {
+        _buffer.write(prefixOtherLines.trimRight());
+        _hasMultipleLines = true;
+      }
+      _buffer.write('\n');
+      _atLineStart = true;
+      return;
+    }
+
+    if (_buffer.isEmpty) {
+      _buffer.write(prefixLineOne);
+    } else if (_atLineStart) {
+      _buffer.write(prefixOtherLines);
+      _hasMultipleLines = true;
+    }
+    bool lineTerminated = false;
+
+    if (s.endsWith('\n')) {
+      s = s.substring(0, s.length - 1);
+      lineTerminated = true;
+    }
+    final List<String> parts = s.split('\n');
+    _buffer.write(parts[0]);
+    for (int i = 1; i < parts.length; ++i) {
+      _buffer
+        ..write('\n')
+        ..write(prefixOtherLines)
+        ..write(parts[i]);
+    }
+
+    if (lineTerminated)
+      _buffer.write('\n');
+
+    _atLineStart = lineTerminated;
+  }
+
+  /// Write text assuming the text already obeys the specified prefixes for the
+  /// first and subsequent lines.
+  void writeRaw(String text) {
+    if (text.isEmpty)
+      return;
+    _buffer.write(text);
+    _atLineStart = text.endsWith('\n');
+  }
+
+
+  /// Write a line assuming the line obeys the specified prefixes. Ensures that
+  /// a newline is added if one is not present.
+  /// The same as [writeRaw] except a newline is added at the end of [line] if
+  /// one is not already present.
+  ///
+  /// A new line is not added if the input string already contains a newline.
+  void writeRawLine(String line) {
+    if (line.isEmpty)
+      return;
+    _buffer.write(line);
+    if (!line.endsWith('\n'))
+      _buffer.write('\n');
+    _atLineStart = true;
+  }
+
+  @override
+  String toString() => _buffer.toString();
+}
+
+/// Defines diagnostics data for an [Object].
+///
+/// DiagnosticsNode provides a high quality multi-line string dump via
+/// [toStringDeep]. The core members are the [name], [header], [getProperties],
+/// [object], and [getChildren]. All other members exist typically to provide
+/// hints for how [toStringDeep] and debugging tools should format output.
+abstract class DiagnosticsNode {
+  DiagnosticsNode({
+    @required this.name,
+    @required this.header,
+    this.style: DiagnosticsTreeStyle.normal,
+    bool hidden: false,
+    this.showNull: false,
+    this.showName: true,
+    this.showSeparator: true,
+    this.emptyDescription,
+    this.defaultValue,
+  })
```

Comment:
so like this?
```dart
 DiagnosticsNode({
    @required this.name,
    @required this.header,
    this.style: DiagnosticsTreeStyle.normal,
    bool hidden: false,
    this.showNull: false,
    this.showName: true,
    this.showSeparator: true,
    this.emptyDescription,
    this.defaultValue,
  }) : _hidden = hidden {
    // A name ending with ':' indicates that the user forgot that the ':' will
    // be automatically added for them when generating descriptions of the
    // property.
    assert(name == null || !name.endsWith(':'));
  }
```

I'm confused as it looks like the flutter style guide seems to imply this is the right style but maybe I'm reading it wrong.
```dart
  Bar({
    Key key,
    Widget childWidget,
  })
      : child = childWidget,
        super(
          key: key,
        );
  final Widget child;
  // ...
}
```

Anyway, just let me know and I'll cleanup all the constructors.